src/shogun/mathematics/linalg/SGLinalg.h

```diff
@@ -99,9 +98,23 @@ class SGLinalg
 	template <class T>
 	T dot(BaseVector<T> *a, BaseVector<T> *b) const;
 
-	/** Check whether gpubackend is registered by user */
+	/** Check whether gpubackend is registered by user
+	 * @return whether gpubackend is registered by user
+	 */
 	bool hasGPUBackend() const;
 
+	/** Wrap SGVector with class CPUVector and GPUVector
+	 * @return pointer of BaseVector
+	 */
+	template <class T>
+	std::shared_ptr<BaseVector<T>> convert_vector(const SGVector<T> &vector) const;
```

Comment:
Maybe call it `get_gpu_type`. It is not very clear from the name that you want to `convert` it into gpu type. Also, we want this to be overloaded so that the users just use the same name while converting both matrix and vector. Maybe you have a better name in mind?